src/main/scala/mesosphere/marathon/api/akkahttp/v2/AppsController.scala

```diff
@@ -324,7 +325,30 @@ class AppsController(
 
   private def listRunningTasks(appId: PathId)(implicit identity: Identity): Route = ???
 
-  private def killTasks(appId: PathId)(implicit identity: Identity): Route = ???
+  private def killTasks(appId: PathId)(implicit identity: Identity): Route = {
+    // the line below doesn't look nice but it doesn't compile if we use parameters directive
+    (forceParameter & parameter("host") & extractTaskKillingMode) {
+      (force, host, mode) =>
+        def findToKill(appTasks: Seq[Instance]): Seq[Instance] = {
+          appTasks.filter(_.agentInfo.host == host || host == "*")
+        }
+        mode match {
+          case TaskKillingMode.Scale =>
+            val deploymentPlan = taskKiller.killAndScale(appId, findToKill, force)
+            onSuccess(deploymentPlan) { plan =>
+              complete((StatusCodes.OK, List(Headers.`Marathon-Deployment-Id`(plan.id)), DeploymentResult(plan.id, plan.version.toOffsetDateTime)))
+            }
+          case TaskKillingMode.Wipe =>
+            onSuccess(taskKiller.kill(appId, findToKill, wipe = true)) { instances =>
+              complete(Json.obj("tasks" -> instances))
```

Comment:
As I mentioned in the comment before. I think the is a RAML type. If not, could you please create one?